opencanary/config.py

```diff
@@ -93,52 +99,19 @@ def getVal(self, key, default=None):
                 return default
             raise e
 
-    def setValues(self, params):  # noqa: C901
+    def checkValues(self):  # noqa: C901
         """Set all the valid values in params and return a list of errors for invalid"""
-
-        # silently ensure that node_id and mac are not modified via web
-        sacred = ["device.node_id", "device.mac"]
-        for k in sacred:
-            if k in params:
-                del params[k]
-
-        # if dhcp is enabled, ignore the static ip settings
-        if params.get("device.dhcp.enabled", False):
-            static = [
-                "device.ip_address",
-                "device.netmask",
-                "device.gw",
-                "device.dns1",
-                "device.dns2",
-            ]
-            for k in static:
-                if k in params:
-                    del params[k]
-
-        # for each section, if disabled, delete ignore section's settings
-        disabled_modules = tuple(
-            filter(
-                lambda m: not params.get("%s.enabled" % m, False),
-                ["ftp", "ssh", "smb", "http"],
-            )
-        )
-        for k in params.keys():
-            if not k.endswith("enabled") and k.startswith(disabled_modules):
-                del params[k]
-                continue
-
+        params = self.__config
         # test options indpenedently for validity
         errors = []
-        for key, value in iteritems(params):
+        for key, value in params.items():
             try:
-                self.valid(key, value)
+                self.is_valid(key, value)
             except ConfigException as e:
                 errors.append(e)
 
         # Test that no ports overlap
-        ports = {k: v for k, v in iteritems(self.__config) if k.endswith(".port")}
-        newports = {k: v for k, v in iteritems(params) if k.endswith(".port")}
-        ports.update(newports)
+        ports = {k: int(v) for k, v in iteritems(params) if k.endswith(".port")}
```

Comment:
Might as well drop `iteriterms` from `six` and use `params.items()`

Reply:
thanks done!